eclair-node/src/main/scala/fr/acinq/eclair/api/ExtraDirectives.scala

```diff
@@ -50,10 +53,19 @@ trait ExtraDirectives extends Directives {
   }
 
   def withChannelIdentifier: Directive1[Either[ByteVector32, ShortChannelId]] = formFields(channelIdFormParam.?, shortChannelIdFormParam.?).tflatMap {
-    case (None, None) => reject(MalformedFormFieldRejection("channelId/shortChannelId", "Must specify either the channelId or shortChannelId"))
     case (Some(channelId), None) => provide(Left(channelId))
     case (None, Some(shortChannelId)) => provide(Right(shortChannelId))
-    case _ => reject(MalformedFormFieldRejection("channelId/shortChannelId", "Must specify either the channelId or shortChannelId"))
+    case _ => reject(MalformedFormFieldRejection("channelId/shortChannelId", "Must specify either the channelId or shortChannelId (not both)"))
+  }
+
+  def withChannelsIdentifier: Directive1[List[Either[ByteVector32, ShortChannelId]]] = formFields(channelIdFormParam.?, channelIdsFormParam.?, shortChannelIdFormParam.?, shortChannelIdsFormParam.?).tflatMap {
+    case (None, None, None, None) => reject(MalformedFormFieldRejection("channelId(s)/shortChannelId(s)", "Must specify channelId, channelIds, shortChannelId or shortChannelIds"))
+    case (channelId_opt, channelIds_opt, shortChannelId_opt, shortChannelIds_opt) =>
+      val channelId: List[Either[ByteVector32, ShortChannelId]] = channelId_opt.map(cid => Left(cid)).toList
+      val channelIds: List[Either[ByteVector32, ShortChannelId]] = channelIds_opt.map(_.map(cid => Left(cid))).toList.flatten
+      val shortChannelId: List[Either[ByteVector32, ShortChannelId]] = shortChannelId_opt.map(scid => Right(scid)).toList
+      val shortChannelIds: List[Either[ByteVector32, ShortChannelId]] = shortChannelIds_opt.map(_.map(scid => Right(scid))).toList.flatten
+      provide(channelId ++ channelIds ++ shortChannelId ++ shortChannelIds)
```

Comment:
Is it worth to call `.distinct` on the resulting collection of ids? We can prevent some user errors in case of repeated ids but not all errors because they can mix `channelId` and `shortChannelId` for the same channel.

Reply:
Good idea, I'll add a `distinct` it doesn't hurt.

Users can indeed mess up and send the same command twice to the same channel if they specify it both via `channelId` and `shortChannelId`, but I figured it was mostly harmless (the channels should handle the duplicate command without any issue) and something the user should fix on their side of the script.

Reply:
Done in https://github.com/ACINQ/eclair/pull/1440/commits/75fd29f9229c6c3666c3fac11df59f5225dc1907